internal/connectors/plugins/public/column/utils.go

```diff
@@ -0,0 +1,239 @@
+	"fmt"
+	"time"
+
+	"github.com/formancehq/payments/internal/connectors/plugins/public/column/client"
+	"github.com/formancehq/payments/internal/models"
+)
+
+func (p *Plugin) validateTransferRequest(pi models.PSPPaymentInitiation) error {
+	if pi.Amount == nil {
+		return fmt.Errorf("required field amount must be provided")
+	}
+
+	if pi.Asset == "" {
+		return fmt.Errorf("required field asset must be provided")
+	}
+
+	if pi.Metadata == nil {
+		return fmt.Errorf("required field metadata must be provided")
+	}
+
+	if pi.SourceAccount == nil {
+		return fmt.Errorf("required field sourceAccount is missing")
+	}
+
+	if pi.SourceAccount.Name == nil {
+		return fmt.Errorf("required sourceAccount field name is missing")
+	}
+
+	if pi.SourceAccount.Reference == "" {
+		return fmt.Errorf("required sourceAccount field reference is missing")
+	}
+
+	if pi.DestinationAccount == nil {
+		return fmt.Errorf("required field destinationAccount must be provided")
+	}
+
+	if pi.DestinationAccount.Reference == "" {
+		return fmt.Errorf("required destinationAccount field reference is missing")
+	}
+
+	allowOverdraft := models.ExtractNamespacedMetadata(pi.Metadata, client.ColumnAllowOverdraftMetadataKey)
+	if allowOverdraft != "" {
+		if allowOverdraft != "true" && allowOverdraft != "false" {
+			return fmt.Errorf("required field allow overdraft must be provided")
+		}
+	}
+
+	hold := models.ExtractNamespacedMetadata(pi.Metadata, client.ColumnHoldMetadataKey)
+	if hold != "" {
+		if hold != "true" && hold != "false" {
+			return fmt.Errorf("required field hold must be provided")
+		}
+	}
+
+	err := validateAddress(pi.Metadata)
+
+	if err != nil {
+		return err
+	}
+
+	addressLine1 := models.ExtractNamespacedMetadata(pi.Metadata, client.ColumnAddressLine1MetadataKey)
+
+	if addressLine1 != "" {
+		countryCode := models.ExtractNamespacedMetadata(pi.Metadata, client.ColumnCountryCodeMetadataKey)
+
+		if countryCode == "" {
+			return fmt.Errorf("required metadata field %s is missing", client.ColumnCountryCodeMetadataKey)
+		}
+
+	}
+
+	if addressLine1 == "" && models.ExtractNamespacedMetadata(pi.Metadata, client.ColumnCountryCodeMetadataKey) != "" {
+		return fmt.Errorf("metadata field %s is not required when addressLine1 is not provided", client.ColumnCountryCodeMetadataKey)
+	}
+
+	return nil
+}
+
+func validateAddress(addressMetadata map[string]string) error {
+	addressLine1 := models.ExtractNamespacedMetadata(addressMetadata, client.ColumnAddressLine1MetadataKey)
+
+	if addressLine1 != "" {
+		city := models.ExtractNamespacedMetadata(addressMetadata, client.ColumnCityMetadataKey)
+
+		if city == "" {
+			return fmt.Errorf("required metadata field %s is missing", client.ColumnCityMetadataKey)
+		}
+	}
+
+	if addressLine1 == "" && models.ExtractNamespacedMetadata(addressMetadata, client.ColumnAddressLine1MetadataKey) != "" {
+		return fmt.Errorf("metadata field %s is not required when addressLine1 is not provided", client.ColumnAddressLine1MetadataKey)
+	}
```

Comment:
Since all the checks below rely on addressLine1 being empty, it would be preferable to return early here so that the if statements later can be simplified (easier to read) and also just so the reader can more easily keep the different logic branches of logic in mind.